src/js/containers/state/StateContainer.jsx

```diff
@@ -35,18 +35,24 @@ export class StateContainer extends React.Component {
 
         this.request = null;
     }
-    componentWillMount() {
+
+    componentDidMount() {
         this.loadStateOverview(this.props.params.stateId, this.props.stateProfile.fy);
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (this.props.params.stateId !== nextProps.params.stateId) {
+    shouldComponentUpdate(nextProps) {
```

Comment:
This seems to be breaking the page. I believe, since you aren't checking `nextState`, the loading message never goes away. We might not need this check at all, what specifically are you trying to block from updating?